rotkehlchen/tasks/assets.py

```diff
@@ -17,6 +36,25 @@
     'common_asset_details AS C ON C.identifier = B.identifier JOIN assets as A on '
     'A.identifier=B.identifier WHERE B.PROTOCOL IS NOT ? ORDER BY C.symbol'
 )
+MULTISEND_SPAM_THRESHOLD = 5
+
+
+def _add_spam_asset(
+        detected_spam_assets: Sequence[str],
+        globaldb: GlobalDBHandler,
+        user_db: DBHandler,
+        user_db_write_ctx: DBCursor,
+) -> None:
+    """Update the protocol to SPAM_PROTOCOL for the list of assets provided and also ignores it"""
+    with globaldb.conn.write_ctx() as write_cursor:
+        query = 'UPDATE evm_tokens SET protocol=? WHERE identifier=?'
+        write_cursor.executemany(
+            query,
+            [(SPAM_PROTOCOL, identifier) for identifier in detected_spam_assets],
+        )
+
+    for asset_id in detected_spam_assets:
+        user_db.add_to_ignored_assets(write_cursor=user_db_write_ctx, asset=Asset(asset_id))
```

Comment:
we have had this discussion before. This is really bad and heavy code. Instead use `add_to_ignored_action_ids` which was made to avoid the resolution and lookup in the DB for every entry of a for loop

```suggestion
    user_db.add_to_ignored_action_ids(write_cursor=user_db_write_cursor, detected_spam_assets)
```

Reply:
As mentioned in discord Asset doesn't query the globaldb. I have created a new method that is a `executemany`